crates/storage/provider/src/providers/mod.rs

```diff
@@ -270,6 +270,16 @@ where
         self.database.history_by_block_hash(block_hash)
     }
 
+    fn state_by_block_hash(&self, block: BlockHash) -> Result<StateProviderBox<'_>> {
+        // check tree first
```

Comment:
Should this also take the txpool into account? e.g. what does eth_storageAt  with tag = `Pending` use?